src/Reducer/data-reducer.js

```diff
@@ -0,0 +1,141 @@
+import { videoExists } from "../Components/Video/videoUtil";
+
+export const dataReducer = (state, { type, payload }) => {
+  switch (type) {
+    /* Video List */
+    case "SET_VIDEOLIST":
+      return { ...state, videoList: payload };
+
+    /* Liked videos */
+    case "TOGGLE_LIKE":
+      return {
+        ...state,
+        toastMsg: videoExists(state.likedVideos, payload)
+          ? "Removed from Liked videos"
+          : "Added to Liked videos",
+        likedVideos: videoExists(state.likedVideos, payload)
+          ? state.likedVideos.filter((video) => video !== payload)
+          : state.likedVideos.concat(payload),
+      };
+
+    /* Toggle videos from existing playlist */
+    case "TOGGLE_PLAYLIST":
+      const list = state.playlist.find(
+        (item) => item.listId === payload.listId
+      );
+      const videoFlag = list.videos.some((videoId) => videoId === payload.id);
```

Comment:
I think videoFlag can be replaced with more precise name

Reply:
videoExists could be a better naming here, however I have already defined a util function with this name.
Will think of something descriptive.